tests/52user-directory/01public.pl

```diff
@@ -5,21 +5,32 @@
       my ( $user ) = @_;
 
       my $room_id;
+      my $searching_user;
 
       my $displayname = generate_random_displayname();
+      my $searching_displayname = generate_random_displayname();
 
       matrix_set_displayname( $user, $displayname )
       ->then( sub {
+         matrix_create_user_on_server( $user->http,
+            displayname => $searching_displayname
+         );
+      })->then( sub {
+         ( $searching_user ) = @_;
+
          matrix_create_room( $user,
             preset => "public_chat",
          );
+      })->then( sub {
+         my ( $room_id ) = @_;
+         matrix_join_room( $searching_user, $room_id );
       })->then( sub {
          ( $room_id ) = @_;
```

Comment:
this looks bogus. join_room doesn't return a room id. (looks like the function-level $room_id is unused and should be vaped?)

Reply:
done